src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java

```diff
@@ -307,5 +302,20 @@ public ResponseEntity<InputStreamResource> downloadCommonStats(
                 .body(isr);
     }
 
+    public CommonsPlus toCommonsPlus(Commons c) {
+        Optional<Integer> numCows = commonsRepository.getNumCows(c.getId());
+        Optional<Integer> numUsers = commonsRepository.getNumUsers(c.getId());
 
+        return CommonsPlus.builder()
+                .commons(c)
+                .totalCows(numCows.orElse(0))
+                .totalUsers(numUsers.orElse(0))
+                .build();
+    }
+
```

Comment:
I suggest adding this Javadoc comment.   The code, as written, has a bit of a code smell to it, but it will at least work.  If we acknowledge that, maybe a future team will be able to figure out a better way to implement this.

```suggestion
/** This method should be called each time a commons object is being used before the value of effective capacity is retrieved from that commons.  This probably should go in the getter for effectiveCapacity, but we don't have access there to the commonsRepository to do the lookup of numUsers.
*/
```

Reply:
I agree, I'll update this